airbyte-integrations/connectors/source-square/unit_tests/test_component.py

```diff
@@ -109,16 +109,15 @@ def test_substream_incremental_sync(state, last_record, expected, expected_strea
         parent_stream=parent_stream,
     )
 
-    actual_stream_slice = next(slicer.stream_slices(SyncMode, state)) if records else {}
+    slicer.set_initial_state(state)
+    actual_stream_slice = next(slicer.stream_slices()) if records else {}
 
     # Covers the test case for abnormal state that is greater than the current time
     if "updated_at" in state and state["updated_at"] > datetime.now().strftime(DATETIME_FORMAT):
         assert actual_stream_slice["updated_at"] != state["updated_at"]
-        slicer.update_cursor(stream_slice=actual_stream_slice, last_record=last_record)
-        assert slicer.get_stream_state()["updated_at"] < state["updated_at"]
     else:
         assert actual_stream_slice == expected_stream_slice
-        slicer.update_cursor(stream_slice=actual_stream_slice, last_record=last_record)
+        slicer.update_state(stream_slice=actual_stream_slice, last_record=last_record)
```

Comment:
The previous test never actually updated the state using `state` but only using `actual_stream_slice` which didn't have the abnormally high `updated_at=2200-01-01T00:00:00.000000Z`. Therefore, this assertion didn't test anything